### PR TITLE
Jan2022

### PR DESCRIPTION
**Pull request by kruisard**
Final versions of simulation labs used in 2022 January session. Significant re-writing of lab exercises and exposition was done between summer and winter session.

### CODE REVIEW

**Comment by kruisard** [comment]
Transverse-dynamics is about the same, some updated plots and refined explanations. Longitudinal dynamics got a whole new section. Dispersion-chromaticity is about the same, with some edits to elegant simulation. FODO-transport got renamed to quadrupole-focusing and includes a new section about matching.

File: Dispersion-Chromaticity.ipynb [hunk 1]
```diff
@@ -43,73 +45,90 @@
...
-    "- Replace 2 meter drift (D2) with two 0.75 meter drifts (D3). \n",
-    "- Sandwich a dipole between the two difts (0.5 m, 20 deg dipole - element CSBEND):\n",
-    "![alt text](images/dipole.png)\n",
-    "- You can add your new element in by dragging and dropping the name from the right hand side into the left hand side.\n",
-    "- Under Tab: Control -> twiss_output, Change Matched to Yes (page 1). \n",
-    "- Go to Visualization tab at top of the page.\n",
-    "- Change simulation to run over 1 cell (simulate beamline “FODOcell” in the drop down box). \n",
+    "- Delete the 0.5 meter drift \"D3\" and, in its place, add the RBEN element \"DIPO\" \n",
+    "- \"DIPO\" is a 20-degree dipole with rectangular geometry (as opposed to sector geometry, which is also an option).\n",
+    "- Run simulation over modified FODO cell\n",
```
Comment (nneveu) [inline Dispersion-Chromaticity.ipynb:64]: add period here?

File: Dispersion-Chromaticity.ipynb [hunk 2]
```diff
@@ -181,43 +233,211 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "## This cell includes the function that makes the plot below. You don't need to look at it, just execute it once.\n",
+    "\n",
```
Comment (nneveu) [inline Dispersion-Chromaticity.ipynb:283]: Don't need to look at what? the function? Maybe reword this to -> You don't need to edit the function, just execute it?

File: Dispersion-Chromaticity.ipynb [hunk 3]
```diff
@@ -181,43 +233,211 @@
...
+    "                    if np.sign(m[j]) > 0:\n",
+    "                        plt.plot([[qxmin]*p.size,[qxmax]*p.size],[p/n2[j] - np.array(m[j]*qxmin/n2[j]), p/n2[j] - np.array(m[j]*qxmax/n2[j])],color=color[i-1],linestyle=linestyle);\n",
+    "                    else:\n",
+    "                        plt.plot([[qxmin]*p.size,[qxmax]*p.size],[p/n1[j] - np.array(m[j]*qxmin/n1[j]), p/n1[j] - np.array(m[j]*qxmax/n1[j])],color=color[i-1],linestyle=linestyle);\n",
+    "                # resonance dif lines\n",
+    "                if lines[3]:\n",
+    "                    if np.sign(m[j]) > 0:\n",
+    "                        plt.plot([[qxmin]*p.size,[qxmax]*p.size],[p/n1[j] - np.array(m[j]*qxmin/n1[j]), p/n1[j] - np.array(m[j]*qxmax/n1[j])],color=color[i-1],linestyle=linestyle);\n",
+    "                    else:\n",
+    "                        plt.plot([[qxmin]*p.size,[qxmax]*p.size],[p/n2[j] - np.array(m[j]*qxmin/n2[j]), p/n2[j] - np.array(m[j]*qxmax/n2[j])],color=color[i-1],linestyle=linestyle);\n",
```
Comment (nneveu) [inline Dispersion-Chromaticity.ipynb:368]: Maybe consider reducing the number of repeated code with another function, or variables that set some values prior to the if statement, since the only changing part is n1/n2?

Comment (nneveu) [inline Dispersion-Chromaticity.ipynb:368]: Maybe consider reducing the number of repeated code with another function, or variables that set some values prior to the if statement, since the only changing part is n1/n2?

File: Longitudinal-Dynamics.ipynb [hunk 4]
```diff
@@ -83,93 +110,111 @@
...
+    "**Q2) We call the boundary between well-behaved elliptal orbits about the synchronous particle and untrapped motion that you found in Q1 the 'separatrix'. What happens to the particles that find themselves outside of this separatrix – are they lost from the synchrotron?** \n",
     "\n",
     "\n",
-    "**Q3) How does the motion behave near the “unstable fixed points”?**\n",
+    "**Q3) For particles that are still within, but very close to the separatrix, how many turns does it take to complete one revolution in the phase space? Use the initial conditions $(\\phi, \\Delta E) = (0,\\Delta E_{max})$ where $\\Delta E_{max}$ is your answer to Q1. In $(\\phi, \\Delta E)$ coordinates, where are the stable and unstable fixed points?**\n",
     "\n",
     "\n",
     "---\n",
-    "Change the synchronous phase, $\\phi_s$, to $30^{\\circ}$ (what happens in a synchrotron when this phase change occurs?).  \n",
+    "Change the synchronous phase, `phi_s`, to `\\phi_s = 30` degrees. Consider what happens in a synchrotron when this phase change occurs.  \n",
```
Comment (nneveu) [inline Longitudinal-Dynamics.ipynb:194]: Extra forward slash (\) before phi_s = 30 is not needed.

File: Longitudinal-Dynamics.ipynb [hunk 5]
```diff
@@ -83,93 +110,111 @@
...
+    "Next, set `initial_deltaE` back to `[2,4,6,8,10]` and change `nturns1` to 1000, leave the other parameters as they are, and change the energy of the synchronous particle, $E_s$, from 8 GeV to 150 GeV.  \n",
+    "\n",
+    "**Q7) By changing the values in `initial_phases` locate the stable phase region - where is the stable fixed point?.**  \n",
+    "(Note: The transition energy of the Main Injector is about 22 GeV; if you've covered this in lecture - what happens at transition?)\n",
     "\n",
     "\n",
     "\n",
-    "**Q7) What maximum $\\Delta E$ will the RF system support now?**\n",
-    "Hint - you will need to find the optimal phase (phi_vals1) and increase the number of turns.\n",
+    "**Q8) What maximum $\\Delta E$ will remain within the RF bucket of the synchronous particle now?**\n",
```
Comment (nneveu) [inline Longitudinal-Dynamics.ipynb:216]: Extra forward slash again for \Delta.

File: Longitudinal-Dynamics.ipynb [hunk 6]
```diff
@@ -178,17 +223,19 @@
...
-    "- Increase the voltage by a factor of 10 and look at the phase space; \n",
-    "- another factor of 10; another; and so on. \n",
+    "Since the difference equations are of a non-linear system, one can push the parameters until resonance islands, chaos, and other non-linear effects are found. \n",
+    "\n",
+    "- Go back to the “stationary bucket” case `phi_s1  = 0`\n",
+    "- Restore beam energy to initial value, `E_s1 = 8*GeV`\n",
+    "- Since increasing the RF voltage increases the accepted range of `\\Delta_E`, it is better to initialize orbits in `phi` instead. Set `initial_deltaE = [0,0,0,0,0]` and `initial_phases = [40,80,120,160,179]`.\n",
+    "- It will be helpful to fix the plot axes. Do this by un-commenting the line `plt.xlim([-200,200]); plt.ylim([-500,500])` in the cell above.\n",
+    "- Increase the voltage by a factor of 10 (by setting `rf_voltage_factor = 10`) and look at the phase space; \n",
+    "- Another factor of 10; another; and so on. \n",
```
Comment (nneveu) [inline Longitudinal-Dynamics.ipynb:236]: I would spell this out like in the bullet right above -> "Increase the voltage by another factor of 10; another; and so on."

File: Transverse-Dynamics.ipynb [hunk 7]
```diff
@@ -2,42 +2,44 @@
  "cells": [
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "tags": []
+   },
    "source": [
-    "![alt text](images/uspas.png)\n",
+    "![alt text](uspas.png)\n",
```
Comment (nneveu) [inline Transverse-Dynamics.ipynb:9]: This image doesn't render on my local copy w/o the images folder.

**Review by nneveu — APPROVED** [review]
I left a few comments, but nothing serious. Nice work & thanks!
